platform/darwin/src/MGLNetworkConfiguration.m

```diff
@@ -34,7 +41,8 @@ + (instancetype)sharedManager {
         _sharedManager = [[self alloc] init];
     });
 
-    [self setNativeNetworkManagerDelegateToDefault];
+    // Notice, this is reset for each call.
+    [MGLNativeNetworkManager sharedManager].delegate = _sharedManager;
```

Comment:
What is the purpose of resetting delegate property? Maybe you can point to code that creates a new sharedManager?

Reply:
Added additional comments.